models/concerns/distance_matrix.rb

```diff
@@ -63,7 +63,7 @@ def compute_need_matrix(&block)
         # set matrix_time and matrix_distance depending of dimensions order
         log "matrix computation #{matrix_points.size}x#{matrix_points.size}"
         tic = Time.now
-        router_matrices = OptimizerWrapper.router.matrix(OptimizerWrapper.config[:router][:url], mode, dimensions, matrix_points, matrix_points, options)
+        router_matrices = router.matrix(OptimizerWrapper.config[:router][:url], mode, dimensions, matrix_points, matrix_points, options)
```

Comment:
Why don't we set the url inside the router object like we do for the `api_key` when we are creating the router object -- something like 
```
OptimizerWrapper.router(OptimizerWrapper.config[:router][:api_key], OptimizerWrapper.config[:router][:url])
```
instead of
```
OptimizerWrapper.router(OptimizerWrapper.config[:router][:api_key])
```

That way we wouldn't have to do `OptimizerWrapper.config[:router][:url]` everywhere. What do you think?

Reply:
the router_wrapper.rb file is the same for 3 projects (it should be moved one day in a gem), the url cannot be different for the moment in none of these 3 projects

Reply:
Yeah it was just an idea, in that case of course it can be done for all projects (if it makes sense). 

All I meant that the router object can handle its own url and the matrix function doesn't really need the url as an argument. Maybve during the passage to the gem, the public interface of the router object can be improved :shrug: